### PR TITLE
Multi file upload

### PR DESCRIPTION
**Pull request by michaelmerrill**
Hey Jay,

I started working on the multiple file component. It's a wip, but I thought I'd check in and see if this is the direction you were thinking? I used rethinkdb as a dummy db since it's been fairly popular with the graphql community lately, but I'm happy to change it.

### CODE REVIEW

File: api/schema.graphql [hunk 1]
```diff
@@ -13,10 +13,10 @@ input Upload {
...
 
 type Query {
-  # GraphQL will not work without defining a query.
-  ignore: Boolean
+  allUploads: [File]
 }
 
 type Mutation {
   singleUpload (file: Upload!): File!
+  multiUpload (files: [Upload!]): [File!]
```
Comment (jaydenseric) [inline api/schema.graphql]: It should be `[Upload!]!` and `[File!]!` to indicate the array itself, as well as the inner items are required.

File: client/components/multi-uploader.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    }
+  }
+
+  render () {
+    return <input type='file' accept={'image/jpeg,image/png'} multiple required onChange={this.handleChange} />
+  }
+}
+
+export default graphql(gql`
+  mutation multiUpload ($files: [Upload!]) {
```
Comment (jaydenseric) [inline client/components/multi-uploader.js]: Same here: `[Upload!]!`.

File: api/config.js [hunk 3]
```diff
@@ -2,3 +2,8 @@ import path from 'path'
 
 export const distPath = path.resolve(__dirname, 'dist')
 export const apiEndpoint = '/graphql'
+export const rethinkdb = {
```
Comment (jaydenseric) [inline api/config.js]: Lets populate these from the `.env` file, and add these suggested values to `.env.example`.

File: api/resolvers.js [hunk 4]
```diff
@@ -1,13 +1,31 @@
...
+  return result.changes.map((file) => file.new_val)
+}
+
 export default {
   Query: {
-    ignore () {
-      return null
+    async allUploads () {
+      const db = getRethinkDB()
+      return await db.table('uploads')
```
Comment (jaydenseric) [inline api/resolvers.js]: Pretty sure Apollo can handle promises and the `await` is not necessary.

File: api/rethinkdb.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import rethinkdbdash from 'rethinkdbdash'
+import {rethinkdb} from './config'
```
Comment (jaydenseric) [inline api/rethinkdb.js]: Perhaps skip config for this, and use the new .env entries directly in this file.

File: app/pages/index.js [hunk 6]
```diff
@@ -13,10 +14,20 @@ export default withData(props => (
         body {
           margin: 2em;
         }
+        section {
+          height: calc(50vh - 4em);
```
Comment (jaydenseric) [inline app/pages/index.js]: Not sure why there is a fixed height, and with a calc? You can do something like `padding-top: 3em;` to create responsive spacing between sections.

File: app/pages/index.js [hunk 7]
```diff
@@ -13,10 +14,20 @@ export default withData(props => (
...
+        }
       `}</style>
     </Head>
-    <h1>Apollo upload example</h1>
-    <p>Select an image to upload and view the response in the console.</p>
-    <SingleUploader />
+    <section>
+      <h1>Apollo upload example (single file)</h1>
```
Comment (jaydenseric) [inline app/pages/index.js]: Lets have a h1 `Apollo upload examples` above the sections, and simplify the text of the section headings and make them h2.

**Review by jaydenseric — CHANGES_REQUESTED** [review]
Awesome work!!! 🙌

**Comment by michaelmerrill** [comment]
Sweet, thanks for the changes. I also added the list component and I'm now getting the `window is undefined` error. Also, did you want to show the images from the /tmp folder? I wasn't quite sure how to do that so instead I just have the names listed.

**Comment by jaydenseric** [comment]
Just the names listed will be fine for now. And great chicken-or-the egg question about the error! Perhaps I will use your WIP, fix that error, and then we merge.

**Comment by jaydenseric** [comment]
Need to add the DB stuff to `.env.example`.

**Comment by jaydenseric** [comment]
I think I have solved all the issues with `apollo-upload-server` and `apollo-upload-client`, but I can't get your fork to work to test it.

I'm getting a lot of `Fail to create a new connection for the connection pool.` errors. Does RethinkDB have to be globally setup or something? If so, we will need to look at a different lightweight DB that does not require any global setup.

**Comment by jaydenseric** [comment]
Quick Google:

- https://github.com/typicode/lowdb
- https://github.com/arvindr21/diskdb

**Comment by michaelmerrill** [comment]
You're right, rethink requires a global setup. Oops! I swapped it out with lowdb. I also updated the apollo-upload-client & server and everything seems to be working. Thank you!

File: api/db.json [hunk 8]
```diff
@@ -0,0 +1 @@
+{"uploads":[{"name":"IMG_0599.JPG","type":"image/jpeg","size":343845,"path":"/tmp/uploads/upload_7b8d12d21774c726904d6272f6176d57"},{"name":"IMG_0599.JPG","type":"image/jpeg","size":343845,"path":"/tmp/uploads/upload_e39a69250284e0113b276e3dcf87dbb4"},{"name":"kalen-emsley-99660.jpg","type":"image/jpeg","size":14036175,"path":"/tmp/uploads/upload_fe22084971ed01e6849cd7fa253ccf4b"}]}
```
Comment (jaydenseric) [inline api/db.json]: Need to add this file to the gitignore, and remove what was already committed here.

**Review by jaydenseric — APPROVED** [review]

**Comment by jaydenseric** [comment]
Ok to merge?

**Comment by michaelmerrill** [comment]
Yep! Changes made and it's gtg! Thanks, this has been fun. Really liked working with your code base 👍

**Comment by jaydenseric** [comment]
Thanks for the help :)

**Pull request merged**